examples/00_basic/app.py

```diff
@@ -1,60 +1,121 @@
-    return 'This is the home page'
-
-
-# creating decorator
-# can be used for email auth
-def token_existence(func):
+#creating decorator
+#can be used for email auth
+def token_existance(func):
     @wraps(func)
-    def checking(*args, **kwargs):
+    def checking(*args,**kwargs):
         token = request.args.get('token')
-        if not token:
-            return jsonify({'message':  'Missing token'}), 403
+        if not token or token in blacklist:
+            return jsonify({'message':  'Missing token'}),403
         try:
             data = jwt.decode(token, app.config['SECRET_KEY'])
+            verified_user = User.query.filter_by(id = data['id']).first()
         except:
-            return jsonify({'message': 'token expired'}), 403
-        return func(*args, **kwargs)
+            return jsonify({'message': 'token is invalid'}),403
+        return func(verified_user,*args, **kwargs)
     return checking
 
+@app.route('/',methods=['GET','POST'])
+def index():
+    return 'Welcome to chatbox'
+
+#go to user home
+@app.route('/home/<token>',methods=['GET','POST'])
+@token_existance
+def user_home(current_user):
+    if current_user.active == "True":
+        return render_template("user_home.html",token=token)
 
+#create/register a user
+@app.route('/register',methods=['GET','POST'])
 def sign_up():
-    pass
-    # create new user
-    # add to database
-    # token can be formed uniquely
+    data = request.form
+
+    #securing the password
+    passw = sc.hash(data['password'])
+   
+    #create new user
+    new_user = User(id=uuid.uuid4(), name = data['name'],active = "False",password = passw, email = data['email'],username = data['username'])
+   
+    #add to database
+    db.session.add(new_user)
+    db.session.commit()
+    
+    return '<h1>New User Created</h1>'
 
+#login to user section
+@app.route('/login',methods=['GET','POST','PUT'])
+def login():
+    auth = request.authorization
 
-def encode_token():
-    # get username
-    # username is unique
-    user = User.query.filter_by(name=username)  # code for sql database
-    token = jwt.encode
+    if not auth or not auth.username or not auth.password:
+        return make_response('verfication failed',401)
+
+    user = User.query.filter_by(username = auth.username).first()
 
     if not user:
-        return make_response('Could not verify', 401)
-    else:
-        token = jwt.encode({'public_id': user.public_id, 'exp': datetime.datetime.utcnow() +
-                datetime.timedelta(minutes=30)}, app.config['SECRET_KEY'])
-        return jsonify({'token': token.decode('UTF-8')})
+        return make_response('Either username or Password is incorrect. Please try Again!!!',401)
+
+    if sc.verify(auth.password,user.password):
+        user.active = "True"
+        tokn= jwt.encode(
+            {
+                'id':user.id,
+                'exp': datetime.datetime.utcnow() + datetime.timedelta(hours=1)
+            },
+            app.config['SECRET']
```

Comment:
I looked over `jwt.encode` quickly but I'm just checking. This key isn't stored in the JWT right? Just used for encryption?